explicitly call python2

MartinLuethi opened the pull request:
Newer Linux distributions have symlinked /usr/bin/python3 to /usr/bin/python,
such that "/usr/bin/env python" calls python3 instead of python2.

Changes:
- replace "/usr/bin/env python" with "/usr/bin/env python2" in all *.py files

- this change choses explicitly the python version (currently python2)

- this patch enables the gradual transition to python3, since python3 scripts
  can be marked with "/usr/bin/env python3"

- it is unlikely that this patch breaks anything on Linux or MacOS X, since
  /usr/bin/env should find the corresponding python version.

Commit:
explicitly call python2, since python might link to python3

Rationale:
Newer Linux distributions have symlinked /usr/bin/python3 to /usr/bin/python,
such that "/usr/bin/env python" calls python3 instead of python2.

Changes:
- replace "/usr/bin/env python" with "/usr/bin/env python2" in all *.py files

- this change choses explicitly the python version (currently python2)

- this patch enables the gradual transition to python3, since python3 scripts
  can be marked with "/usr/bin/env python2"

- it is unlikely that this patch breaks anything on Linux or MacOS X, since
  /usr/bin/env should find the corresponding python version.

closes  #11090

brianmoose commented:
Looks like you rebased your branch against master which is why you have all those merge commits.
You'll want to rebase against the devel branch and repush.

MartinLuethi commented:
Not sure if something went wrong. I rebased on devel, and did the pull request with respect to devel.
Everything like described on the web page. 
Should I try a new pull request?

brianmoose commented:
You probably don't need a new PR, you just need to fix up your branch. I guess you added your commit to a branch based of master. You need to add it to a branch based on devel.
Don't know the quickest way, but this might work (assuming you have the "upstream" remote):
```
git checkout env_python2 # start in the current branch
git checkout -b env_python2-orig # make a copy in case something goes wrong
git fetch upstream # get up to date with idaholab/moose
git checkout -b tmp-branch upstream/devel # create a temporary branch based on idaholab/moose:devel
git cherry-pick dfebf5a # put your commit onto the branch
git branch -f -m tmp-branch env_python2 # overwrite the old branch with the temporary branch
git push -f origin env_python2 # force push back to github.
```
If something goes horribly wrong then you still have your commit in the `env_python2-orig` branch

MartinLuethi commented:
thanks a lot for your detailed instructions
I now did exactly that, hope the commit is correct now.

Best, Martin

brianmoose commented on modules/combined/doc/moosedocs.py:
```diff
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python222222222222222222
```
bad python

brianmoose commented on python/peacock/BasePeacockMainWindow.py:
```diff
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python22
```
bad python in all the `python/peacock` files

brianmoose commented on test/tools/large_subdomain_gen/_output_cubit_commands.py:
```diff
@@ -1,4 +1,4 @@
-#!/usr/bin/env python2.5
+#!/usr/bin/env python2.52
```
bad python

brianmoose commented on test/tools/large_subdomain_gen/large_sub_gen.py:
```diff
@@ -1,4 +1,4 @@
-#!/usr/bin/env python2.5
+#!/usr/bin/env python2.52
```
bad python

andrsd commented:
Yeah, or just:
```
git rebase --onto devel HEAD~1 env_python2
```
(assuming devel has the current HEAD)

Commit:
explicitly call python2, since python might link to python3

Rationale:
Newer Linux distributions have symlinked /usr/bin/python3 to /usr/bin/python,
such that "/usr/bin/env python" calls python3 instead of python2.

Changes:
- replace "/usr/bin/env python" with "/usr/bin/env python2" in all *.py files

- this change choses explicitly the python version (currently python2)

- this patch enables the gradual transition to python3, since python3 scripts
  can be marked with "/usr/bin/env python2"

- it is unlikely that this patch breaks anything on Linux or MacOS X, since
  /usr/bin/env should find the corresponding python version.

closes  #11090

MartinLuethi commented:
Sorry, I should have checked all files by hand, which I did now. Some were changed several times, maybe due to symlinks pointing to the same file.

Anyway, the changes should be clean now, /usr/bin/env python2 everywhere.

permcody commented:
Martin, Nice meeting you this week. Thanks for the PR!

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/169170/) on e51bcf8 wanted to post the following:

View the site [here](http://mooseframework.inl.gov/docs/PRs/11091/site)

This comment will be updated on new commits.

permcody reviewed (approved):
Martin, Thanks for the contribution.
